climpred/bootstrap.py

```diff
@@ -498,88 +498,136 @@ def bootstrap_compute(
+        for i in range(iterations):
+            # resample with replacement
+            smp_hind = _resample(hind, resample_dim)
+            # compute init skill
+            init_skill = compute(
+                smp_hind,
                 verif,
-                alignment=alignment,
                 metric=metric,
                 comparison=comparison,
-                dim=dim,
+                alignment=alignment,
                 add_attrs=False,
+                dim=dim,
                 **metric_kwargs,
             )
-        )
-        # compute persistence skill
-        # impossible for probabilistic
-        if not metric.probabilistic:
-            pers.append(
-                reference_compute(
-                    smp_hind,
+            # reset inits when probabilistic, otherwise tests fail
+            if (
+                resample_dim == 'init'
+                and metric.probabilistic
+                and 'init' in init_skill.coords
+            ):
+                init_skill['init'] = hind.init.values
+            bootstrapped_init_skill.append(init_skill)
+            # generate uninitialized ensemble from hist
+            uninit_hind = resample_uninit(hind, hist)
+            # compute uninit skill
+            bootstrapped_uninit_skill.append(
+                compute(
+                    uninit_hind,
                     verif,
+                    alignment=alignment,
                     metric=metric,
-                    alignment=reference_alignment,
+                    comparison=comparison,
+                    dim=dim,
                     add_attrs=False,
                     **metric_kwargs,
                 )
             )
-    init = xr.concat(init, dim='iteration', **CONCAT_KWARGS)
-    uninit = xr.concat(uninit, dim='iteration', **CONCAT_KWARGS)
-    # when persistence is not computed set flag
-    if pers != []:
-        pers = xr.concat(pers, dim='iteration', **CONCAT_KWARGS)
+            # compute persistence skill
+            # impossible for probabilistic
+            if not metric.probabilistic:
+                pers_skill.append(
+                    reference_compute(
+                        smp_hind,
+                        verif,
+                        metric=metric,
+                        alignment=reference_alignment,
+                        add_attrs=False,
+                        **metric_kwargs,
+                    )
+                )
+        bootstrapped_init_skill = xr.concat(
+            bootstrapped_init_skill, dim='iteration', **CONCAT_KWARGS
+        )
+        bootstrapped_uninit_skill = xr.concat(
+            bootstrapped_uninit_skill, dim='iteration', **CONCAT_KWARGS
+        )
+        if pers_skill != []:
+            bootstrapped_pers_skill = xr.concat(
+                pers_skill, dim='iteration', **CONCAT_KWARGS
+            )
+            pers_output = True
+        else:
+            pers_output = False
+
+    else:  # if resample_dim == 'member':
```

Comment:
this in the new stuff. first bootstrapping, then metric. later on and in the classes we can tear those calls apart